addons/purchase_stock/models/res_partner.py

```diff
@@ -17,24 +18,31 @@ class ResPartner(models.Model):
 
     @api.depends('purchase_line_ids')
     def _compute_on_time_rate(self):
-        order_lines = self.env['purchase.order.line'].search([
+        order_line_ids = self.env['purchase.order.line']._search([
             ('partner_id', 'in', self.ids),
```

Comment:
`('partner_id', 'child_of', self.commercial_partner_id.ids),`
Create a supplier A, with two child partner C, D
Create order with partner C
Create order with partner D, the on_time_rate is -1